driver-core/src/main/com/mongodb/event/EventListenerSettings.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright 2017 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.event;
+
+import com.mongodb.annotations.NotThreadSafe;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.util.Collections.unmodifiableList;
+
+/**
+ * Settings for the various event listeners.
+ *
+ * @since 3.5
+ */
+public final class EventListenerSettings {
```

Comment:
What's the reasoning behind the addition of this class and corresponding deprecation of the listener properties already present in ClusterSettings and ServerSettings.  
Without that change, ConnectionPoolSettings is an obvious home for ConnectionPoolListener. 
That just leaves ConnectionListener.  That could go in SocketSettings, though that's one level too low in the current architecture.  Or, ConnectionListener could just be removed.  It doesn't offer much more than the connectionAdded and connectionRemoved events in ConnectionPoolListener.  The only thing that would be missed are connections for SDAM heartbeats.